lib/TokenManager.php

```diff
@@ -294,4 +294,8 @@ public function setGuestName(Wopi $wopi, ?string $guestName = null): Wopi {
 	public function getUrlSrc(File $file): string {
 		return $this->wopiParser->getUrlSrcValue($file->getMimeType());
 	}
+
+	public function getUrlSrcForMimeType(string $type): string {
+		return $this->wopiParser->getUrlSrcValue($type);
```

Comment:
I think this may throw an exception. Maybe we rather catch that and return a nullable string? Then the frontend check in the settings code also makes sense `this.initial.setting_iframe_url ?? ''`